In [1]:
import os
import json

### NO resampling

### Find total num of subjects and file path for creating config.json later 

In [9]:
# Get current working directory
current_dir = os.getcwd()

# Go into "data" folder and list all entries
app_dir = current_dir+"/data"
entries = os.listdir(app_dir)

# Initialize an empty list to store folder path and subjectID
folders = []
subjectID = []

# Iterate over all entries
LENGTH = len("sub-")
for entry in entries:
    # Check if the entry is a directory
    if os.path.isdir(os.path.join(app_dir, entry)):
        # If it is, add it to the list of folders
        subjectID.append(int(entry[LENGTH:]))
        folders.append("../data/" + entry + "/cortexmap")

# sort folder names alphabetically
folders.sort()

# Print num of folder to verify
print("Total number of folders: ", len(folders))

Total number of folders:  1062


### Create config.json file for fsaverage app

In [10]:
# Create a dictionary with "input" key (threshold and dilate set to 0)
json_data = {"threshold" : 0,
             "dilate" : 0,
             "inputs": folders}

# Specify the output JSON file path (outputs to current directory)
output_file_path = 'config.json'

# Write data to JSON file
with open(output_file_path, 'w') as f:
    json.dump(json_data, f, indent=4)

print(f"Data has been written to {output_file_path}")

Data has been written to config.json


### RANDOM NUM GENERATOR - don't run now

### Find total num of subjects and file path

In [ ]:
import numpy as np

In [6]:
# Get current working directory
current_dir = os.getcwd()

# List all entries in the directory
entries = os.listdir(current_dir)

# Initialize an empty list to store folder path and subjectID
folders = []
subjectID = []

# Iterate over all entries
LENGTH = len("sub-")
for entry in entries:
    # Check if the entry is a directory
    if os.path.isdir(os.path.join(current_dir, entry)):
        # If it is, add it to the list of folders
        subjectID.append(int(entry[LENGTH:]))
        folders.append("../data_indiv_cortexmap" + entry + "/cortexmap")
# sort folder names alphabetically
folders.sort()

# Print num of folder to verify
print("Total number of folders: ", len(folders))

Folders in the current directory: ['../data_indiv_cortexmapsub-196750/cortexmap', '../data_indiv_cortexmapsub-126628/cortexmap', '../data_indiv_cortexmapsub-397154/cortexmap', '../data_indiv_cortexmapsub-134627/cortexmap', '../data_indiv_cortexmapsub-175035/cortexmap', '../data_indiv_cortexmapsub-174841/cortexmap', '../data_indiv_cortexmapsub-136227/cortexmap', '../data_indiv_cortexmapsub-571144/cortexmap', '../data_indiv_cortexmapsub-748662/cortexmap', '../data_indiv_cortexmapsub-627852/cortexmap', '../data_indiv_cortexmapsub-169747/cortexmap', '../data_indiv_cortexmapsub-140925/cortexmap', '../data_indiv_cortexmapsub-486759/cortexmap', '../data_indiv_cortexmapsub-499566/cortexmap', '../data_indiv_cortexmapsub-148436/cortexmap', '../data_indiv_cortexmapsub-311320/cortexmap', '../data_indiv_cortexmapsub-685058/cortexmap', '../data_indiv_cortexmapsub-153429/cortexmap', '../data_indiv_cortexmapsub-151728/cortexmap', '../data_indiv_cortexmapsub-626648/cortexmap', '../data_indiv_cortexmaps

### Generate boostrap samples

- Number of subjects = 1062
- Number of bootstrap samples = 1000
- So, there are 1000 samples with 1062 input files in each sample (subjects sampled with replacement)

In [20]:
initial_seed = 3212024121403 # can be any random number

# Generate random seeds
rng = np.random.default_rng(initial_seed)
num_samples = 1000 # num of bootstrap samples = num of seeds
seeds = rng.choice(1000000000000000000, num_samples, replace=False) # includes 1 seed per sample

# Generate bootstrap samples
samples = [] # 2D arr. 1 row = 1 boostrap sample. 1 element = 1 subject = index folders arr

for seed in seeds:
    rng = np.random.default_rng(seed)
    total_subjects = len(folders) # total number of subjects = size of sample
    samples.append(rng.choice(total_subjects, total_subjects))

# print dimension of samples to verify
print(len(samples))
print(len(samples[0]))

1000
15


### Create config.json file for fsaverage app

THINK about order of file names

In [23]:
# Create list of folder paths for 1 boostrap sample
sample_index = 0 # index of which bootstrap sample
subjects = [] # folder paths for 1 boostrap sample
for i in samples[sample_index]:
    subjects.append(folders[i])

# Create a dictionary with "input" key (threshold and dilate set to 0)
json_data = {"threshold" : 0,
             "dilate" : 0,
             "inputs": subjects}

# Specify the output JSON file path (outputs to current directory)
output_file_path = 'config.json'

# Write data to JSON file
with open(output_file_path, 'w') as f:
    json.dump(json_data, f, indent=4)

print(f"Data has been written to {output_file_path}")

Data has been written to config.json
